server.go

```diff
@@ -1504,6 +1507,60 @@ func (c *createMeasurementsIfNotExistsCommand) addFieldIfNotExists(measurement,
 	return nil
 }
 
+func (s *Server) applyDropSeries(m *messaging.Message) error {
+	var c dropSeriesCommand
+	mustUnmarshalJSON(m.Data, &c)
+
+	database := s.databases[c.Database]
+	if database == nil {
+		return ErrDatabaseNotFound
+	}
+
+	seriesByMeasurement := make(map[string][]uint32)
+
+	// Remove from metastore.
+	for _, seriesID := range c.SeriesIDs {
+		series := database.series[seriesID]
+		if series == nil {
+			return fmt.Errorf("series not found for id %d", seriesID)
+		}
+		measurement := series.measurement
+		if measurement == nil {
+			return fmt.Errorf("measurement not found for series %d", seriesID)
+		}
+		seriesByMeasurement[measurement.Name] = append(seriesByMeasurement[measurement.Name], seriesID)
+	}
+
+	err := s.meta.mustUpdate(m.Index, func(tx *metatx) error {
+		if err := tx.dropSeries(c.Database, seriesByMeasurement); err != nil {
+			return err
+		}
+
+		// Delete series from the database.
+		if err := database.dropSeries(c.SeriesIDs...); err != nil {
```

Comment:
This dropSeries should also be updated to take the `seriesByMeasurement` map. That way you're not looping through every single measurement and trying to drop it